### PR TITLE
NSFW module, search service, and additional fixes

### PR DESCRIPTION
**Pull request by Ivy-Wusky**
some code taken from old versions of [nadekobot](https://gitlab.com/Kwoth/nadekobot)

### CODE REVIEW

**Comment by Ivy-Wusky** [comment]
also don't worry it checks if the channel is an nsfw channel or not

**Comment by Ivy-Wusky** [comment]
> i t i s n o t C A P S

indeed

**Comment by Ivy-Wusky** [comment]
e621 now supported btw

**Comment by LoserIsPoggers** [comment]
Hi I see comment box so I make comment wait will everyone see this?

**Comment by Ivy-Wusky** [comment]
> Hi I see comment box so I make comment wait will everyone see this?

i think so idk

**Comment by LoserIsPoggers** [comment]
> > Hi I see comment box so I make comment wait will everyone see this?
> 
> i think so idk

Oh shit well hi

File: IvyBot/IvyBot/Services/NSFWService.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+namespace IvyBot.Services
+{
+    public class NSFWService
+    {
+        public static async Task<string> GetRule34File(string tag)
+        {
+            try
+            {
+                var random = new Random();
+                var url = $"http://rule34.xxx/index.php?page=dapi&s=post&q=index&limit=100&tags={tag.Replace(" ", "_")}";
```
Comment (Grass0001) [inline IvyBot/IvyBot/Services/NSFWService.cs:15]: 🔫

File: IvyBot/IvyBot/Services/NSFWService.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            catch (Exception ex)
+            {
+                // Console.WriteLine(ex.Message);
+                return $"Error in rule34 search: {ex.Message}";
+            }
+        }
+
+        public static async Task<string> GetE621File(string tag)
+        {
+            try
```
Comment (Grass0001) [inline IvyBot/IvyBot/Services/NSFWService.cs:34]: this is the word try

File: IvyBot/IvyBot/Services/SearchService.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                    {
+                        formdatcontent = new FormUrlEncodedContent(headers);
+                    }
+                    
+                    var message = await something.PostAsync(url, formdatcontent).ConfigureAwait(false);
+                    return await message.Content.ReadAsStreamAsync().ConfigureAwait(false);
+                
+                default:
+                    
+                    throw new NotImplementedException("Request type unsupported.");
```
Comment (Grass0001) [inline IvyBot/IvyBot/Services/SearchService.cs:57]: please don't throw, this is a no throw zone

File: IvyBot/IvyBot/Services/SearchService.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            }
+        }
+        
+        public static async Task<string> GetResponseStringAsync(string url, IEnumerable<KeyValuePair<string, string>> headers = null, RequestHttpMethod method = RequestHttpMethod.GET)
+        {
+            using (var streampoop = new StreamReader(await GetResponseStreamAsync(url, headers, method).ConfigureAwait(false)))
```
Comment (Grass0001) [inline IvyBot/IvyBot/Services/SearchService.cs:63]: ah yes, "streampoop"

File: README.md [hunk 5]
```diff
@@ -25,7 +25,7 @@ clone the repository with `git clone https://github.com/Ivy-Wusky/ivy-bot.git` t
 
 remember that if you are running your **lavalink** instance on [heroku](https://www.heroku.com) the port will always be **80** regardless of the port you set
 # future plans
-additional commands and an **nsfw** module
```
Comment (Grass0001) [inline README.md:28]: rip(?)

**Review by Grass0001 — APPROVED** [review]
i must say the number 42, have some spam review

File: IvyBot/IvyBot/Services/NSFWService.cs (hunk 2 above)
Comment (Ivy-Wusky) [reply IvyBot/IvyBot/Services/NSFWService.cs:34]: yes

File: IvyBot/IvyBot/Services/SearchService.cs (hunk 3 above)
Comment (Ivy-Wusky) [reply IvyBot/IvyBot/Services/SearchService.cs:57]: i have a throw pass, i can throw anywhere

File: IvyBot/IvyBot/Services/SearchService.cs (hunk 4 above)
Comment (Ivy-Wusky) [reply IvyBot/IvyBot/Services/SearchService.cs:63]: this is me when i don’t know what to call variables

File: README.md (hunk 5 above)
Comment (Ivy-Wusky) [reply README.md:28]: no(?)

File: IvyBot/IvyBot/Services/NSFWService.cs (hunk 1 above)
Comment (Ivy-Wusky) [reply IvyBot/IvyBot/Services/NSFWService.cs:15]: no u

**Comment by Ivy-Wusky** [comment]
gib more nsfw command ideas

**Comment by Ivy-Wusky** [comment]
you know what fuck it

**Pull request merged**